core/src/main/java/org/elasticsearch/index/query/RegexpQueryBuilder.java

```diff
@@ -59,35 +156,47 @@ public RegexpQueryBuilder flags(RegexpFlag... flags) {
                 value |= flag.value;
             }
         }
-        this.flags = value;
+        this.flagsValue = value;
+        return this;
+    }
+
+    public RegexpQueryBuilder flags(int flags) {
+        this.flagsValue = flags;
         return this;
     }
 
+    public int flags() {
+        return this.flagsValue;
+    }
+
     /**
      * Sets the regexp maxDeterminizedStates.
      */
     public RegexpQueryBuilder maxDeterminizedStates(int value) {
         this.maxDeterminizedStates = value;
-        this.maxDetermizedStatesSet = true;
```

Comment:
while refactoring some of the queries I did find quite a few bugs just like that. Should we always backport them to master? What's our overall policy on this matter?